### PR TITLE
Add logs to ISA

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: client/src/api/block_builder/input_selection/core/remainder.rs [hunk 1]
```diff
@@ -131,15 +133,17 @@ impl InputSelection {
...
         }
 
         let Some((remainder_address, chain)) = self.get_remainder_address() else {
-                return Err(Error::MissingInputWithEd25519Address);
+            return Err(Error::MissingInputWithEd25519Address);
         };
 
         // TODO checked ops ?
-        let mut remainder_builder = BasicOutputBuilder::new_with_amount(inputs_sum - outputs_sum)?;
+        let diff = inputs_sum - outputs_sum;
```
Comment (kwek20) [inline client/src/api/block_builder/input_selection/core/remainder.rs:145]: 🤯

File: client/src/api/block_builder/input_selection/core/remainder.rs [hunk 2]
```diff
@@ -150,6 +154,8 @@ impl InputSelection {
 
         let remainder = remainder_builder.finish_output(self.protocol_parameters.token_supply())?;
 
+        log::debug!("Created remainder output of {diff} for {remainder_address:?}");
```
Comment (Thoralf-M) [inline client/src/api/block_builder/input_selection/core/remainder.rs:158]: Could the NTs also be interesting?

File: client/src/api/block_builder/input_selection/core/remainder.rs (hunk 1 above)
Comment (thibault-martinez) [reply client/src/api/block_builder/input_selection/core/remainder.rs:145]: Well I need to log it later so...

Comment (kwek20) [reply client/src/api/block_builder/input_selection/core/remainder.rs:145]: ye i saw after haha

**Review by kwek20 — DISMISSED** [review]

**Review by kwek20 — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**